source/system/native/FileUnix.ooc

```diff
@@ -27,313 +30,219 @@ readdir_r: extern func (DIR*, DirEnt*, DirEnt**) -> Int
+		match res {
+			case 0 => (result st_mode & S_IXUSR) != 0
+			case => false
 		}
-
-		/**
-		 * set the executable bit on this file's permissions for
-		 * current user, group, and other.
-		 */
-		setExecutable: override func (exec: Bool) -> Bool {
-			lstatResult, fstatResult: FileStat
-			res := lstat(this path as CString, lstatResult&)
-			if (res != 0) return false // couldn't get file mode
-			fd := open(this path as CString, READWRITE)
-			if (fd == -1) return false
-			res = fstat(fd, fstatResult&)
-			if (res != 0) {
-				if(close(fd) != 0)
-					raise("fstat failed, close(fd) failed")
-				return false
-			}
-			if (fstatResult st_ino != lstatResult st_ino || fstatResult st_mode != lstatResult st_mode) {
-				if(close(fd) != 0)
-					raise("fstat != lstat, close(fd) failed")
-				return false
-			}
-			mode := fstatResult st_mode
-			if (exec)
-				mode |= (S_IXUSR | S_IXGRP | S_IXOTH)
-			else
-				mode &= ~(S_IXUSR | S_IXGRP | S_IXOTH)
-			result := fchmod(fd, mode) == 0
+	}
+	setExecutable: override func (exec: Bool) -> Bool {
+		lstatResult, fstatResult: FileStat
+		res := lstat(this path as CString, lstatResult&)
+		if (res != 0)
+			return false // couldn't get file mode
+		fd := open(this path as CString, READWRITE)
+		if (fd == -1)
+			return false
+		res = fstat(fd, fstatResult&)
+		if (res != 0) {
+			if(close(fd) != 0)
+				raise("fstat failed, close(fd) failed")
+			return false
+		}
+		if (fstatResult st_ino != lstatResult st_ino || fstatResult st_mode != lstatResult st_mode) {
 			if (close(fd) != 0)
-				result = false
-			result
-		}
-
-		/**
-		 * @return the time of last access, or -1 if it doesn't exist
-		 */
-		lastAccessed: override func -> Long {
-			result: FileStat
-			res := lstat(this path as CString, result&)
-			match res {
-				case 0 => result st_atime as Long
-				case => -1
-			}
-		}
-
-		/**
-		 * @return the time of last modification, or -1 if it doesn't exist
-		 */
-		lastModified: override func -> Long {
-			result: FileStat
-			res := lstat(this path as CString, result&)
-			match res {
-				case 0 => result st_mtime as Long
-				case => -1
-			}
-		}
-
-		/**
-		 * @return the time of creation, or -1 if it doesn't exist
-		 */
-		created: override func -> Long {
-			result: FileStat
-			res := lstat(this path as CString, result&)
-			match res {
-				case 0 => result st_ctime as Long
-				case => -1
-			}
-		}
-
-		/**
-		 * @return true if the function is relative to the current directory
-		 */
-		relative: override func -> Bool {
-			// that's a bit rough, but should work most of the time
-			!this path startsWith("/")
+				raise("fstat != lstat, close(fd) failed")
+			return false
+		}
+		mode := fstatResult st_mode
+		if (exec)
+			mode |= (S_IXUSR | S_IXGRP | S_IXOTH)
+		else
+			mode &= ~(S_IXUSR | S_IXGRP | S_IXOTH)
+		result := fchmod(fd, mode) == 0
+		if (close(fd) != 0)
+			result = false
+		result
+	}
+	lastAccessed: override func -> Long {
```

Comment:
Maybe we should change these functions to return `DateTime` instead ? Not this PR of course.